opslevel/resource_opslevel_webhook_action.go

```diff
@@ -0,0 +1,174 @@
+package opslevel
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opslevel/opslevel-go/v2022"
+)
+
+func resourceWebhookAction() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages a webhook action",
+		Create:      wrap(resourceWebhookActionCreate),
+		Read:        wrap(resourceWebhookActionRead),
+		Update:      wrap(resourceWebhookActionUpdate),
+		Delete:      wrap(resourceWebhookActionDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the webhook action.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "The description of the webhook action.",
+				ForceNew:    false,
+				Optional:    true,
+			},
+			"liquid_template": {
+				Type:        schema.TypeString,
+				Description: "Template that can be used to generate a webhook payload.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"webhook_url": {
+				Type:        schema.TypeString,
+				Description: "The URL of the webhook action.",
+				ForceNew:    false,
+				Required:    true,
+			},
+			"http_method": {
+				Type:         schema.TypeString,
+				Description:  "The http method used to call the webhook action.",
+				ForceNew:     false,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(opslevel.AllCustomActionsHttpMethodEnum(), false),
+			},
+			"headers": {
+				Type:        schema.TypeMap,
+				Description: "HTTP headers to be passed along with your webhook when triggered.",
+				ForceNew:    false,
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceWebhookActionCreate(d *schema.ResourceData, client *opslevel.Client) error {
+
+	input := opslevel.CustomActionsWebhookActionCreateInput{
+		Name:           d.Get("name").(string),
+		Description:    opslevel.NewString(d.Get("description").(string)),
+		LiquidTemplate: d.Get("liquid_template").(string),
+		WebhookURL:     d.Get("webhook_url").(string),
+		HTTPMethod:     opslevel.CustomActionsHttpMethodEnum(d.Get("http_method").(string)),
+		Headers:        convertHeadersMap(d.Get("headers")),
+	}
+
+	resource, _ := client.CreateWebhookAction(input)
+	//if err != nil {
```

Comment:
Couldn't figure out what was going on here, was getting an invalid syntax error from somewhere but it was still creating the webhook action successfully.  Now it's still creating successfully but it was failing here with a can't Unmarshal 'Accept' field or something to that effect. Had issues with this the whole time though and spent too much time trying to figure it out so just commented it out to be able to move on.